Drivers/Inc/Display.h

```diff
@@ -1,48 +1,80 @@
-/* Copyright (c) 2020 UT Longhorn Racing Solar */
-
-/* Driver for the display board. This communicates
- * with the Nextion display over UART in order to show
- * critical information to the driver.
- */
+/** Copyright (c) 2022 UT Longhorn Racing Solar
+ * @file Display.c
+ * @brief Function prototypes for the display driver.
+ * 
+ * This contains function prototypes relevant to sending/receiving messages
+ * to/from our Nextion HMI.
+ * 
+ * @author Ishan Deshpande (IshDeshpa)
+ * @author Roie Gal (Cam0Cow)
+*/
 
 #ifndef __DISPLAY_H
 #define __DISPLAY_H
 
-#include "common.h"
-#include "config.h"
+#include "common.h"	// common headers
+#include "config.h"	// for ErrorStatus
+#include "Tasks.h"	// for os and fault error locs
+
+#define MAX_ARGS 2	// maximum # of arguments in a command packet
 
-/*
- * Initialize the Nextion display
+/**
+ * All three pages on the HMI 
  */
-void Display_Init();
+typedef enum{
+	STARTUP=0,
+	INFO,
+	FAULT
+} Page_t;
 
 /**
- * Value setting subroutines
+ * For display elements with three states
  */
-ErrorStatus Display_SetMainView(void);
+typedef enum{
+	DISABLED=0,
+	ENABLED,
+	ACTIVE
+} TriState_t;
 
-ErrorStatus Display_CruiseEnable(State on);
-ErrorStatus Display_SetVelocity(float vel);
-ErrorStatus Display_SetSBPV(uint16_t mv);
-ErrorStatus Display_SetError(int idx, char *err);
-ErrorStatus Display_NoErrors(void);
+/**
+ * Packages relevant display command data for use in a fifo
+ */
+typedef struct{
+	char* compOrCmd;
+	char* attr;
+	char* op;
+	uint8_t numArgs;
+	bool argTypes[MAX_ARGS];	// TRUE for integers, FALSE for strings
```

Comment:
Might want to have an enum with INT_ARG and STRING_ARG, it'll make the code using this struct a bit clearer.